JavaScript/Topics/8.JSON/README.md

```diff
@@ -0,0 +1,203 @@
+# JSON
+
+
+## What is JSON?
+---
+
+JSON stands for JavaScript Object Notation it is a text format for storing and transporting data which is "self-describing" and easy to understand. The JSON object contains methods for parsing JavaScript Object Notation (JSON) and converting values to JSON. It can't be called or constructed, and aside from its two method properties, it has no interesting functionality of its own.
+
+Example:
+
+```JSON
+'{"name":"Rahul", "age":30, "salary":13000}' //a JSON string
+```
+It defines an object with 3 properties:
+- name
+- age
+- salary
+  
+Each property has a value.
+If we parse the JSON string in JavaScript, we can access the data as an object.
+
+*Note: The JSON syntax is derived from JavaScript object notation, but the JSON format is text only. Code for reading and generating JSON exists in many programming languages.*
+
+## Why use JSON?
+---
+
+The JSON format is syntactically similar to the code for creating JavaScript objects. Because of this, a JavaScript program can easily convert JSON data into JavaScript objects. Since the format is text only, JSON data can easily be sent between computers, and used by any programming language.
+
+## JSON Syntax
+---
+
+JSON Syntax Rules:
+
+- Data is in name/value pairs
+- Data is separated by commas
+- Curly braces hold objects
+- Square brackets hold arrays
+
+### JSON Data
+---
+JSON data is written as name/value pairs (aka key/value pairs). A name/value pair consists of a field name (in double quotes), followed by a colon, followed by a value.
+
+Example:
+
+```JSON
+"name":"Rahul"
+```
+
+### JSON - Evaluates to JavaScript Objects
+---
+The JSON format is almost identical to JavaScript objects. In JSON, keys must be strings, written with double quotes.
+
+Example:
+
+#### JSON
+---
+```JSON
+"name": "Rahul"
+```
+
+In JavaScript, keys can be strings, numbers, or identifier names.
+
+Example:
+
+#### Javascript
+---
+```js
+name: "Rahul"
+```
+
+*Note: JSON names require double quotes.*
+
+### JSON Values
+---
+In JSON, values must be one of the following data types:
+
+- string
+- number
+- object
+- array
+- boolean
+- null
+
+## JSON vs XML
+---
+
+Both JSON and XML can be used to receive data from a web server.
+
+### JSON
+```JSON
+{"students":[
+  { "firstName":"Rahul", "lastName":"Roy" },
+  { "firstName":"Raj", "lastName":"Biswas" },
+  { "firstName":"Anjali", "lastName":"Mathur" }
+]}
+```
+
```

Comment:
Write like this

```
```JSON
{
     "students":
          [
               { "firstName":"Rahul", "lastName":"Roy" },
               { "firstName":"Raj", "lastName":"Biswas" },
               { "firstName":"Anjali", "lastName":"Mathur" }
          ]
}
```